### PR TITLE
`blocking` parameter for Message.js

### PR DESCRIPTION
**Pull request by walaura**
## What does this change?
Pretty much what it says on the tin. Lets a message block the page until action is taken on it to hide it. This will be used in an upcoming sign-in experiment.

I added basic keyboard support. However, the page can scroll freely underneath, which is a bit of an oversight but can be added in a subsequent PR if needed (the specs for the sign-in experiment are still a bit up in the air)

All details are up for debate, let me know what you think!

## Screenshots

screen shot 2018-04-25 at 11 44 26 am
(Terrifying mental picture I know)

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/ui/message.js [hunk 1]
```diff
@@ -50,10 +52,12 @@ class Message {
         this.customJs = opts.customJs || noop;
         this.customOpts = opts.customOpts || {};
         this.$footerMessage = $('.js-footer-message');
+        this.$siteMessage = $('.js-site-message');
     }
 
     show(message: string): boolean {
         const siteMessage = $('.js-site-message');
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/ui/message.js]: Now you've defined `this.$siteMessage` do we need to declare `siteMessage` here and query the dom again when doing so? (I'm assuming the results of `$('.js-site-message')` are the same when we call it in `constructor` and `show`).

Comment (GHaberis) [reply static/src/javascripts/projects/common/modules/ui/message.js]: Yeah, it's not a big issue, but as you made the assumption in `trapFocus()` I figured it'd be safe to do the same in `show()` and just use `this.$siteMessage` there too. 

I'm not a fan of exposing properties by adding them to `this`, but it seem to be the done thing in this module when functions need to share variables as it's been setup as a `Class` (something I'd like to look at changing).

File: static/src/javascripts/projects/common/modules/ui/message.js [hunk 2]
```diff
@@ -143,9 +159,28 @@ class Message {
         );
     }
 
+    trapFocus(): void {
+        const messageEl = this.$siteMessage[0];
+        const [trapStartEl, trapEndEl] = new Array(2).fill(
```
Comment (sndrs) [inline static/src/javascripts/projects/common/modules/ui/message.js]: is this effectively doing

```js
const trapStartEl = document.createElement('div');
const trapEndEl = document.createElement('div');
```
?

Comment (sndrs) [reply static/src/javascripts/projects/common/modules/ui/message.js]: i think so? 

i wouldn't be afraid of repeating yourself here – if you write out explicitly what is happening, the next developer that reads your code should be able to understand your intention immediately. 

as things stand, you need to understand that an Array is being created, filled and destructured (none of which is the specific intention of the code) to understand that you're creating two divs and assigning them to variables (which I think is your intention). 

it's also worth bearing in mind [how this code will be handled by babel](http://babeljs.io/repl/#?babili=false&browsers=%0A&build=&builtIns=usage&code_lz=MYewdgzgLgBA2lATgQwA4GUrMVAogGwBoYk1cwATAgXRgF4YwBTAdxgEFEUBPACgCYAlADoAZgEt8-XgCgAkBRDAArgFsmYKMOCImyKEwJN1m3gHIK4gG5nBMwQG4gA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=true&lineWrap=true&presets=es2015%2Ces2017%2Creact%2Cstage-0%2Cstage-1%2Cstage-2%2Cstage-3&prettier=true&targets=&version=6.26.0&envVersion=1.6.2) (85 bytes becomes 1kb)

Comment (walaura) [reply static/src/javascripts/projects/common/modules/ui/message.js]: holy moly that babel size!

File: static/src/javascripts/projects/common/modules/ui/message.js [hunk 3]
```diff
@@ -72,13 +78,25 @@ class Message {
...
+
         // Add site modifier message
         if (this.cssModifierClass) {
-            siteMessage.addClass(`site-message--${this.cssModifierClass}`);
+            this.$siteMessageContainer.addClass(
+                `site-message--${this.cssModifierClass}`
+            );
         }
 
-        this.$siteMessage = $('.js-site-message__message');
```
Comment (walaura) [inline static/src/javascripts/projects/common/modules/ui/message.js]: whyyy

**Review by GHaberis — APPROVED** [review]
👍

**Pull request merged**